x-pack/platform/plugins/shared/streams_app/public/components/data_management/stream_detail_enrichment/state_management/stream_enrichment_state_machine/persistent_field_mappings_utils.ts

```diff
@@ -0,0 +1,186 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { i18n } from '@kbn/i18n';
+import type { IToasts } from '@kbn/core/public';
+import type { SchemaField, MappedSchemaField } from '../../../schema_editor/types';
+import type { PersistentFieldMapping } from './types';
+
+export interface FieldMatchingResult {
+  exactMatches: FieldRestoration[];
+  potentialReorderings: ReorderingDetection[];
+  unmatchedPersistentMappings: PersistentFieldMapping[];
+}
+
+export interface FieldRestoration {
+  fieldName: string;
+  currentField: SchemaField;
+  persistentMapping: PersistentFieldMapping;
+  matchType: 'exact' | 'position';
+}
+
+export interface ReorderingDetection {
+  fieldName: string;
+  oldPosition: number;
+  newPosition: number;
+  persistentMapping: PersistentFieldMapping;
+  confidence: 'high' | 'medium' | 'low';
+}
+
+export function findExactFieldMatches(
+  currentFields: SchemaField[],
+  persistentMappings: Map<string, PersistentFieldMapping>
+): FieldRestoration[] {
+  const exactMatches: FieldRestoration[] = [];
+
+  currentFields.forEach((field) => {
+    const persistentMapping = persistentMappings.get(field.name);
+    if (persistentMapping) {
+      exactMatches.push({
+        fieldName: field.name,
+        currentField: field,
+        persistentMapping,
+        matchType: 'exact',
+      });
+    }
+  });
+
+  return exactMatches;
+}
+
+export function detectFieldReordering(
```

Comment:
This seems to be "too smart" for me:
* The complexity of the solution is quite high
* It's not clear for the user when it will work and when it won't

I think we should simplify and only remember by the name and re-apply that, nothing else.